Gems/Atom/RHI/Code/Source/RHI/PipelineStateCache.cpp

```diff
@@ -198,15 +200,18 @@ namespace AZ
             if (doesPSODataExist)
             {
                 Ptr<PipelineLibrary> pipelineLibrary = Factory::Get().CreatePipelineLibrary();
-                ResultCode resultCode = pipelineLibrary->Init(*m_device, entry.m_serializedData.get());
+                RHI::PipelineLibraryDescriptor pipelineLibraryDescriptor;
+                pipelineLibraryDescriptor.m_serializedData = entry.m_serializedData.get();
+                pipelineLibraryDescriptor.m_filePath = m_libraryFileNames[handle.GetIndex()];
+                ResultCode resultCode = pipelineLibrary->Init(*m_device, pipelineLibraryDescriptor);
```

Comment:
This code to copy a value from 2 vectors to 1 struct could be saved by just storing the PipelineLibraryDescriptor in m_globalLibrarySet.